test/images/mount-tester/mt.go

```diff
@@ -232,39 +234,32 @@ func readWriteNewFile(path string, perm os.FileMode) error {
 	return readFileContent(path)
 }
 
-func readFileContentInLoop(path string, retryDuration int) error {
+func readFileContentInLoop(path string, retryDuration int, checkInitialContent bool) error {
 	if path == "" {
 		return nil
 	}
-	var content []byte
-	content = testFileContent(path, retryDuration)
-
-	fmt.Printf("content of file %q: %v\n", path, string(content))
-
+	testFileContent(path, retryDuration, checkInitialContent)
 	return nil
 }
 
-func testFileContent(filePath string, retryDuration int) []byte {
+func testFileContent(filePath string, retryDuration int, checkInitialContent bool) {
 	var (
 		contentBytes []byte
 		err          error
 	)
-
 	retryTime := time.Second * time.Duration(retryDuration)
 	for start := time.Now(); time.Since(start) < retryTime; time.Sleep(2 * time.Second) {
 		contentBytes, err = ioutil.ReadFile(filePath)
-		if err == nil {
+		if err != nil {
+			fmt.Printf("Error read file %s: %v, retry\n", filePath, err)
+			continue
+		}
+		if checkInitialContent {
```

Comment:
this variable also seems misnamed. if true, the length is checked every iteration of the loop - not just "initially".